src/Client.cpp

```diff
@@ -297,7 +297,7 @@ void Client::setCustomDataTypes(std::vector<DataType> dataTypes) {
 }
 
 static void setStateCallback(ClientContext& context, ClientState state, StateCallback&& callback) {
-    context.stateCallbacks.at(static_cast<size_t>(state)) = callback;
+    context.stateCallbacks.at(static_cast<size_t>(state)) = std::move(callback);
```

Comment:
As far as I could figure out it is. See [here](https://stackoverflow.com/a/58761212).
In this context, even though this function only accepts rvalue references for `callback`, it has a name and we can take the address from it, both of which are indicators of lvalues. Hence, since it is a named variable, `callback` is an lvalue within the function body. Therefore, `std::move` is necessary to convert it back to an rvalue reference and avoid copying `callback`.